dev_guide/builds.adoc

```diff
@@ -846,42 +851,43 @@ source:
 
-A binary source potentially extracts content (if `*asFile*` is not set), in which case `*contextDir*`
-allows changing to a subdirectory within the content before the build executes.
+With `oc start-build` consider these options:
 
-The source definition is part of the `*spec*` section in the `*BuildConfig*`:
+1) `--from-file`:  The contents of the file you specify will be sent as a binary stream to the builder.  Then
+the builder will store the data in a file with the same name at the top of the build context (this internally maps
+to the `*binary.asFile*` field of the `*Binary*` source type).
 
-====
-[source,yaml]
-----
-source:
-  type: "Binary"
-  binary: <1>
-    asFile: "webapp.war" <2>
-  contextDir: "app/dir" <3>
-  dockerfile: "FROM centos:7\nRUN yum install -y httpd" <4>
-----
-<1> The `*binary*` field specifies the details of the binary source.
-<2> The `*asFile*` field specifies the name of the file that will be created with
-the binary contents.
-<3> The `*contextDir*` field specifies a subdirectory within the extracted contents of a
-binary archive.
-<4> If the optional `*dockerfile*` field is provided, it should be a string
-containing an inline Dockerfile that potentially replaces one within the
-contents of the binary archive.
-====
+2) `--from-dir` and `--from-repo`:  The contents will be archived (either zip, tar, or gzipped tar based on what
+is available on the client system) and sent as a binary stream to the builder.  Then the builder will extract
+the contents of the archive off of the top of the build context.
+
+In each of the above cases:
+
+1) If your `*BuildConfig*` already has a `*Binary*` source type defined, it will effectively be ignored and replaced by what the
+client sends.
+
+2) If your `*BuildConfig*` has a `*Git*` source type defined, it will be dynamically disabled, since `*Binary*` and `*Git*` are
+mutually exclusive, and the data in the binary stream provided to the builder will take precedence.
+
+3) Triggers will be ignored.
+
+When using `oc new-build --binary=true`, the command essentially ensures that any of the restrictions related with binary builds
+are enforced.  And technically source type of `*Binary*` will be created.  But it is really an descriptive placeholder.  You still have to
+use `oc start-build` with one of the "--from" options to actually provide the binary data.
+
+When any form of `*Binary*` build source is being used, the `*dockerfile*` field can optionallly be used.  If used, its contents serve as a replacement Dockerfile to any Dockerfile found in
+the binary stream if the binary stream is an archive.  If the binary stream is a single file, the `*dockerfile*` field contents are stored in a Dockerfile stored at the top of the build context, and it among
+other things can access the file specified by `*binary.asFile*`.  In the case of the binary stream encapsulating extracted archive content, the value of the `*contextDir*` field is interpreted as a subdirectory
```

Comment:
"When any form of `*Binary*` build source is being used, the `*dockerfile*` field can optionallly be used.  If used, its contents serve as a replacement Dockerfile to any Dockerfile found in
the binary stream if the binary stream is an archive, or the single file if the --from-file argument is used and the file argument is named "Dockerfile".  The binary content (either the extracted archive or the single file) will be placed in the context directory used by the Docker build which is performed using the Dockerfile.  

In the case of the binary stream encapsulating extracted archive content, the value of the `*contextDir*` field is interpreted as a subdirectory within the archive, and if valid, the builder changes into that subdirectory before executing the build."